internal/pkg/v2/handler/core/data/event.go

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package data
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/edgexfoundry/edgex-go/internal/core/data/config"
+	model "github.com/edgexfoundry/edgex-go/internal/pkg/v2/go-mod/models/coredata"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/infrastructure/interfaces"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/metadata"
+
+	"github.com/edgexfoundry/go-mod-messaging/messaging"
+	msgTypes "github.com/edgexfoundry/go-mod-messaging/pkg/types"
+)
+
+func AddNewEvent(
+	e model.Event, ctx context.Context,
+	lc logger.LoggingClient,
+	dbClient interfaces.DBClient,
+	chEvents chan<- interface{},
+	msgClient messaging.MessageClient,
+	mdc metadata.DeviceClient,
+	configuration *config.ConfigurationStruct) (string, error) {
+
+	err := checkDevice(e.Device, ctx, mdc, configuration)
+	if err != nil {
+		return "", err
+	}
+
+	if configuration.Writable.ValidateCheck {
+		lc.Debug("Validation enabled, parsing events")
+		//for reading := range e.Readings {
+		//	// Check value descriptor
+		//	name := e.Readings[reading].Name
+		//	vd, err := dbClient.ValueDescriptorByName(name)
+		//	if err != nil {
+		//		if err == db.ErrNotFound {
+		//			return "", errors.NewErrValueDescriptorNotFound(name)
+		//		} else {
+		//			return "", err
+		//		}
+		//	}
+		//	err = isValidValueDescriptor(vd, e.Readings[reading])
+		//	if err != nil {
+		//		return "", err
+		//	}
+		//}
+	}
```

Comment:
Can we remove the above commented code in v1 since we don't have Value Descriptor in v2? If yes, what check should we have?

Reply:
we validate Event in contract, so we can remove this section.

Reply:
Removed.